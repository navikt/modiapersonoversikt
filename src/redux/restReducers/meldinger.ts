import { getMeldinger } from '../../api/meldinger-api';
import { createActionsAndReducer } from './restResource';

const { reducer, action, reload, tilbakestill } = createActionsAndReducer('meldinger');

export function hentMeldinger(fødselsnummer: string) {
    return action(() => getMeldinger(fødselsnummer));
}

export function reloadMeldinger(fødselsnummer: string) {
    return reload(() => getMeldinger(fødselsnummer));
}

export function resetMeldingerResource() {
    return tilbakestill;
}

export default reducer;
