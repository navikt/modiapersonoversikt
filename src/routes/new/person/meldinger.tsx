import { createFileRoute } from '@tanstack/react-router';
import { keepQueryParams } from 'src/utils/keepQueryParams';
import { z } from 'zod';

const meldingerSearchSchema = z.object({
    traadId: z.string().optional()
});

export const meldingerRouteMiddleware = keepQueryParams(['traadId']);

export const Route = createFileRoute('/new/person/meldinger')({
    validateSearch: meldingerSearchSchema,
    search: {
        middlewares: [meldingerRouteMiddleware]
    }
});
