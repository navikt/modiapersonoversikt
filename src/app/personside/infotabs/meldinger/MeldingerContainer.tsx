import * as React from 'react';
import { Traad } from '../../../../models/meldinger/meldinger';
import styled from 'styled-components';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import TraadVisningContainer from './traadvisning/TraadVisningContainer';
import RestResourceConsumer from '../../../../rest/consumer/RestResourceConsumer';
import VerktoylinjeContainer from './traadvisning/verktoylinje/VerktoylinjeContainer';
import TraadListe from './traadliste/TraadListe';
import { CenteredLazySpinner } from '../../../../components/LazySpinner';

const meldingerMediaTreshold = pxToRem(900);

const MeldingerArticleStyle = styled.article`
    @media (min-width: ${meldingerMediaTreshold}) {
        display: flex;
        align-items: flex-start;
        > *:last-child {
            margin-left: ${theme.margin.layout};
        }
    }
    > * {
        margin-bottom: ${theme.margin.layout};
    }
    position: relative;
    > *:first-child {
        flex: 30% 1 1;
    }
    > *:last-child {
        position: sticky;
        top: 0;
        flex: 70% 1 1;
    }
`;

function MeldingerContainer() {
    return (
        <article>
            <RestResourceConsumer<Traad[]>
                getResource={restResources => restResources.tr√•derOgMeldinger}
                returnOnPending={<CenteredLazySpinner />}
            >
                {data => (
                    <MeldingerArticleStyle>
                        <TraadListe traader={data} />
                        <div>
                            <VerktoylinjeContainer />
                            <TraadVisningContainer />
                        </div>
                    </MeldingerArticleStyle>
                )}
            </RestResourceConsumer>
        </article>
    );
}

export default MeldingerContainer;
