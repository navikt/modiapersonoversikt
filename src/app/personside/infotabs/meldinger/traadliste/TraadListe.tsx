import * as React from 'react';
import { useEffect, useRef } from 'react';
import { Traad } from '../../../../../models/meldinger/meldinger';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import styled from 'styled-components/macro';
import theme from '../../../../../styles/personOversiktTheme';
import { Checkbox, Input } from 'nav-frontend-skjema';
import { Normaltekst } from 'nav-frontend-typografi';
import TraadListeElement from './TraadListeElement';
import { LenkeKnapp } from '../../../../../components/common-styled-components';
import SlaaSammenOppgaverKnapp from './besvarflere/SlåSammenOppgaverKnapp';
import usePaginering from '../../../../../utils/hooks/usePaginering';
import { loggEvent } from '../../../../../utils/logger/frontendLogger';
import { guid } from 'nav-frontend-js-utils';
import usePrinter from '../../../../../utils/UsePrinter';
import PrintKnapp from '../../../../../components/PrintKnapp';
import MeldingerPrintMarkup from '../../../../../utils/MeldingerPrintMarkup';
import Panel from 'nav-frontend-paneler';

interface Props {
    traader: Traad[];
    traaderEtterSokOgFiltrering: Traad[];
    valgtTraad: Traad;
    sokeord: string;
    setSokeord: (newSokeord: string) => void;
    skjulVarsler: boolean;
    setSkjulVarsler: (skjul: boolean) => void;
}

const StyledPanel = styled(Panel)`
    ol {
        list-style: none;
    }
`;

const SokVerktøyStyle = styled.div`
    padding: 0 ${theme.margin.layout} ${theme.margin.layout};
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;

const SpaceBetween = styled.div`
    padding: ${theme.margin.layout} ${theme.margin.layout} 0 0;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;
const StyledOl = styled.ol`
    > * {
        border-top: ${theme.border.skille};
    }
    &:focus {
        ${theme.focus};
    }
`;

const InputStyle = styled.div`
    padding: ${theme.margin.layout} ${theme.margin.layout} 0;
    .skjemaelement {
        margin-bottom: 0.2rem;
    }
    .skjemaelement__label {
        ${theme.visuallyHidden};
    }
`;

const PagineringStyling = styled.div`
    padding: 0 ${theme.margin.layout};
    label {
        ${theme.visuallyHidden};
    }
`;

const PrevNextButtonsStyling = styled.div`
    padding: ${theme.margin.layout};
    border-top: ${theme.border.skilleSvak};
`;

const StyledCheckbox = styled(Checkbox)`
    padding: ${theme.margin.layout};
    margin-bottom: 0 !important;
`;

export const valgtMeldingKlasse = 'valgt_melding';

function PrintAlle({ traader }: { traader: Traad[] }) {
    const printer = usePrinter();
    const PrinterWrapper = printer.printerWrapper;

    return (
        <>
            <PrintKnapp tittel={'Skriv ut alle meldinger'} onClick={() => printer?.triggerPrint()} />
            <PrinterWrapper>
                {traader.map(traad => (
                    <MeldingerPrintMarkup key={traad.traadId} valgtTraad={traad} />
                ))}
            </PrinterWrapper>
        </>
    );
}
function TraadListe(props: Props) {
    const inputRef = React.useRef<HTMLInputElement>();
    const paginering = usePaginering(props.traaderEtterSokOgFiltrering, 50, 'melding', props.valgtTraad);
    const sokTittelId = useRef(guid());
    const listeId = useRef(guid());
    const traadListeRef = useRef<HTMLOListElement>(null);

    useEffect(() => {
        const valgtMelding = traadListeRef.current?.getElementsByClassName(valgtMeldingKlasse)[0] as HTMLInputElement;
        valgtMelding?.focus();
    }, [props.valgtTraad]);

    if (props.traader.length === 0) {
        return <AlertStripeInfo>Det finnes ingen meldinger for bruker.</AlertStripeInfo>;
    }

    const visAlleMeldingerKnapp = props.sokeord !== '' && (
        <LenkeKnapp
            onClick={() => {
                props.setSokeord('');
                if (inputRef.current) {
                    inputRef.current.focus();
                }
            }}
        >
            Vis alle meldinger
        </LenkeKnapp>
    );

    const meldingTekst = props.traader.length === 1 ? 'melding' : 'meldinger';
    const soketreffTekst =
        props.traaderEtterSokOgFiltrering.length !== props.traader.length
            ? `Søket traff ${props.traaderEtterSokOgFiltrering.length} av ${props.traader.length} ${meldingTekst}`
            : `Totalt ${props.traader.length} ${meldingTekst}`;

    const onMeldingerSok = (event: React.ChangeEvent<HTMLInputElement>) => {
        const sokeOrd = event.target.value;
        const brukerStarterEtNyttSøk = props.sokeord === '' && sokeOrd.length > 0;
        if (brukerStarterEtNyttSøk) {
            loggEvent('SøkIMeldinger', 'Meldinger');
        }
        props.setSokeord(sokeOrd);
    };

    const handleSkjulVarsler = () => {
        !props.skjulVarsler && loggEvent('SkjulVarsler', 'Meldinger');
        props.setSkjulVarsler(!props.skjulVarsler);
    };

    return (
        <nav aria-label="Velg melding">
            <StyledPanel>
                <SlaaSammenOppgaverKnapp traader={props.traader} />
                <article aria-labelledby={sokTittelId.current}>
                    <h3 id={sokTittelId.current} className="sr-only">
                        Filtrer meldinger
                    </h3>
                    <InputStyle>
                        <Input
                            inputRef={
                                ((ref: HTMLInputElement) => {
                                    inputRef.current = ref;
                                }) as any
                            }
                            value={props.sokeord}
                            onChange={onMeldingerSok}
                            label={'Søk etter melding'}
                            placeholder={'Søk etter melding'}
                            className={'move-input-label'}
                        />
                    </InputStyle>
                    <SpaceBetween>
                        <StyledCheckbox
                            label="Skjul varsler"
                            checked={props.skjulVarsler}
                            onChange={handleSkjulVarsler}
                        />
                        <PrintAlle traader={props.traader} />
                    </SpaceBetween>
                    <SokVerktøyStyle>
                        <Normaltekst aria-live="assertive">{soketreffTekst}</Normaltekst>
                        {visAlleMeldingerKnapp}
                    </SokVerktøyStyle>
                </article>
                <h3 className="sr-only" id={listeId.current}>
                    Meldingsliste - {soketreffTekst}
                </h3>
                {paginering.pageSelect && <PagineringStyling>{paginering.pageSelect}</PagineringStyling>}
                <StyledOl aria-labelledby={listeId.current} tabIndex={-1} ref={traadListeRef}>
                    {paginering.currentPage.map(traad => (
                        <TraadListeElement
                            traad={traad}
                            key={traad.traadId}
                            erValgt={traad === props.valgtTraad}
                            listeId="traadliste-meldinger"
                        />
                    ))}
                </StyledOl>
                {paginering.prevNextButtons && (
                    <PrevNextButtonsStyling>{paginering.prevNextButtons}</PrevNextButtonsStyling>
                )}
            </StyledPanel>
        </nav>
    );
}

export default TraadListe;
