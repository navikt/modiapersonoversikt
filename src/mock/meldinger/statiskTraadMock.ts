import { LestStatus, Meldingstype, Traad } from '../../models/meldinger/meldinger';
import { saksbehandlerTekst } from '../../app/personside/infotabs/meldinger/utils/meldingerUtils';
import { Temagruppe } from '../../models/Temagrupper';

export const statiskTraadMock: Traad = {
    traadId: 'sg838exr',
    meldinger: [
        {
            id: 'x4k72bwd',
            meldingstype: Meldingstype.SAMTALEREFERAT_OPPMOTE,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'nzxo4q', fornavn: 'Sander', etternavn: 'Hansen' }),
            journalfortAv: { ident: 'vlzaiw', fornavn: 'Adrian', etternavn: 'Pettersen' },
            fritekst:
                'Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.',
            lestDato: '2019-07-12',
            status: LestStatus.Lest,
            opprettetDato: '2019-07-30',
            journalfortDato: '2019-07-11',
            ferdigstiltDato: '2019-07-07',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q1i6ndyj',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xmqgb8', fornavn: 'Frida', etternavn: 'Pedersen' }),
            journalfortAv: { ident: 'ly4pps', fornavn: 'Emil', etternavn: 'Nguyen' },
            fritekst:
                'Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.',
            lestDato: '2019-07-14',
            status: LestStatus.IkkeLest,
            opprettetDato: '2019-08-10',
            journalfortDato: '2019-07-21',
            ferdigstiltDato: '2019-07-16',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: 'q57oayxa',
            meldingstype: Meldingstype.SPORSMAL_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'e5otkk', fornavn: 'Synne', etternavn: 'Lunde' }),
            journalfortAv: { ident: '3t55yt', fornavn: 'Victoria', etternavn: 'Andresen' },
            fritekst:
                'Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.',
            lestDato: '2019-08-11',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-06',
            journalfortDato: '2019-08-12',
            ferdigstiltDato: '2019-07-12',
            erFerdigstiltUtenSvar: true,
            erDokumentMelding: false
        },
        {
            id: '90l2nkx0',
            meldingstype: Meldingstype.DELVIS_SVAR_SKRIFTLIG,
            temagruppe: Temagruppe.Arbeid,
            skrevetAvTekst: saksbehandlerTekst({ ident: 'xy70r8', fornavn: 'Kristoffer', etternavn: 'Fredriksen' }),
            journalfortAv: { ident: 'p0jh0d', fornavn: 'Jenny', etternavn: 'Berg' },
            fritekst:
                'Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.',
            lestDato: '2019-07-23',
            status: LestStatus.Lest,
            opprettetDato: '2019-08-13',
            journalfortDato: '2019-07-30',
            ferdigstiltDato: '2019-07-27',
            erFerdigstiltUtenSvar: false,
            erDokumentMelding: false
        }
    ]
};
