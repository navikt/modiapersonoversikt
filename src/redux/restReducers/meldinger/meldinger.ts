import { createRestResourceReducerAndActions } from '../../../rest/utils/restResource';
import { apiBaseUri } from '../../../api/config';
import { AppState } from '../../reducers';
import { Traad } from '../../../models/meldinger/meldinger';

export function getMeldingerFetchUri(state: AppState): string {
    const fnr = state.gjeldendeBruker.f√∏dselsnummer;
    return `${apiBaseUri}/dialog/${fnr}/meldinger`;
}

export default createRestResourceReducerAndActions<Traad[]>('meldinger', getMeldingerFetchUri);
