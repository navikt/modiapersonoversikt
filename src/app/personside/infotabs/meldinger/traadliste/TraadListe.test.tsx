import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../test/Testprovider';
import { getTestStore } from '../../../../../test/testStore';
import TraadListe from './TraadListe';
import { statiskTraadMock } from '../../../../../mock/meldinger/statiskTraadMock';

test('Viser Traadliste', () => {
    const store = getTestStore();
    const traader = [statiskTraadMock];

    const container = renderer.create(
        <TestProvider customStore={store}>
            <TraadListe traader={traader} valgtTraad={traader[0]} traaderEtterSokOgFiltrering={traader} />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
