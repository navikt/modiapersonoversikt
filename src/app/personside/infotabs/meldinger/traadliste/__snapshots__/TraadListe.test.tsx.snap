// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser Traadliste 1`] = `
.c16 {
  position: relative;
}

.c16 svg {
  height: 1.5625rem;
  width: 1.5625rem;
  opacity: 0.4;
}

.c17 {
  position: absolute;
  top: -0.1875rem;
  right: -0.5rem;
  background-color: #78706a;
  color: white;
  height: 1.1rem;
  width: 1.1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 0;
  margin-left: 0.8rem;
  overflow-wrap: break-word;
}

.c18 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c18 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c18 .order-third {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-top: 0.2rem;
}

.c20 > *:not(:last-child) {
  margin-bottom: 0.2rem;
  margin-right: 0.2rem;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 .order-first {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c19 .order-second {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 > *:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c21 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c13 {
  padding: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c12 {
  background-color: #cce1f3;
}

.c12:focus-within {
  position: relative;
  outline: none;
  box-shadow: none;
}

.c12:focus-within::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c12 .hover-animation {
  -webkit-transition: -webkit-transform 0.3s;
  -webkit-transition: transform 0.3s;
  transition: transform 0.3s;
}

.c12:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c12:hover .hover-animation {
  -webkit-transform: translateX(0.5rem);
  -ms-transform: translateX(0.5rem);
  transform: translateX(0.5rem);
}

.c22 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c22 .nav-frontend-chevron {
  width: 1.5625rem;
}

.c4 {
  border: none;
  background-color: transparent;
  padding: 0.1rem 0.2rem;
  border-radius: .35rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #0067c5;
}

.c4 svg {
  margin-left: 0.5rem;
  height: 1.2rem;
  width: 1.2rem;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c4:hover {
  opacity: 0.8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 {
  margin-bottom: 2rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1rem 0rem;
}

.c9 > * {
  margin-bottom: 1rem !important;
}

.c7 {
  margin-bottom: 2rem;
  border-bottom: solid 0.0625rem #78706a;
  page-break-inside: avoid;
}

.c0 ol {
  list-style: none;
}

.c10 {
  padding: 0 .5rem .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  padding: .5rem .5rem 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c11 > * {
  border-top: solid 0.0625rem #78706a;
}

.c11:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c1 {
  padding: .5rem .5rem 0;
}

.c1 .skjemaelement {
  margin-bottom: 0.2rem;
}

.c1 .skjemaelement__label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c3 {
  padding: .5rem;
  margin-bottom: 0 !important;
}

@media print {
  .c4 {
    display: none;
  }
}

@media not print {
  .c5 {
    display: none;
  }
}

<nav
  aria-label="Velg melding"
>
  <div
    className="panel c0"
  >
    <article
      aria-labelledby="Helt tilfeldig ID"
    >
      <h3
        className="sr-only"
        id="Helt tilfeldig ID"
      >
        Filtrer meldinger
      </h3>
      <div
        className="c1"
      >
        <div
          className="skjemaelement move-input-label"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Søk etter melding
          </label>
          <input
            aria-invalid={false}
            className="skjemaelement__input input--fullbredde"
            id="Helt tilfeldig ID"
            onChange={[Function]}
            placeholder="Søk etter melding"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="c2"
      >
        <div
          className="skjemaelement c3"
        >
          <input
            aria-invalid={false}
            checked={false}
            className="skjemaelement__input checkboks"
            id="Helt tilfeldig ID"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Skjul varsler
          </label>
        </div>
        <button
          aria-label="Skriv ut"
          className="c4"
          onClick={[Function]}
        >
          <span
            className="typo-normal"
          >
            Skriv ut alle meldinger
          </span>
           
          <svg
            viewBox="0 0 17 16"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
              fill="#0067C5"
              fillRule="evenodd"
            />
            <path
              d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
              fill="#0067C5"
            />
          </svg>
        </button>
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <p
              className="typo-element"
            >
              Henvendelsen er avsluttet uten å svare bruker
               
               
            </p>
          </div>
          <div
            className="c7"
          >
            <p
              className="typo-ingress"
            >
              Samtalereferat oppmøte - Arbeid
            </p>
            <div
              className="c8"
            >
              <div>
                <p
                  className="typo-element"
                >
                  Kanal: NAV_NO
                </p>
                <p
                  className="typo-element"
                >
                  Type: 
                  SAMTALEREFERAT_OPPMOTE
                </p>
                <p
                  className="typo-element"
                >
                  Temagruppe: 
                  ARBD
                </p>
                <p
                  className="typo-element"
                >
                   Journalført: 
                  11.07.2019
                   
                </p>
                <p
                  className="typo-element"
                >
                  Journalført med team: 
                </p>
              </div>
              <div>
                <p
                  className="typo-element"
                >
                  Fødselsnummer: 
                  10108000398
                </p>
                <p
                  className="typo-normal"
                >
                  Mottatt/Sendt: 
                  07.07.2019
                </p>
                <p
                  className="typo-normal"
                >
                  Lest: 
                  12.07.2019 00:00
                </p>
              </div>
            </div>
            <div
              className="c9"
            >
              <p
                className="typo-element"
              >
                Innhold:
              </p>
              <div>
                <p
                  className="typo-normal"
                >
                  Ea laborum accusamus et ex voluptates provident eveniet voluptatem. Sunt et sed eligendi harum exercitationem. Eaque est maiores autem sunt occaecati atque ipsa ea itaque. Dolorum enim quaerat alias ut sit labore pariatur.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c7"
          >
            <p
              className="typo-ingress"
            >
              Spørsmål fra bruker - Arbeid
            </p>
            <div
              className="c8"
            >
              <div>
                <p
                  className="typo-element"
                >
                  Kanal: NAV_NO
                </p>
                <p
                  className="typo-element"
                >
                  Type: 
                  SPORSMAL_SKRIFTLIG
                </p>
                <p
                  className="typo-element"
                >
                  Temagruppe: 
                  ARBD
                </p>
                <p
                  className="typo-element"
                >
                   Journalført: 
                  12.08.2019
                   
                </p>
                <p
                  className="typo-element"
                >
                  Journalført med team: 
                </p>
              </div>
              <div>
                <p
                  className="typo-element"
                >
                  Fødselsnummer: 
                  10108000398
                </p>
                <p
                  className="typo-normal"
                >
                  Mottatt/Sendt: 
                  12.07.2019
                </p>
                <p
                  className="typo-normal"
                >
                  Lest: 
                  11.08.2019 00:00
                </p>
              </div>
            </div>
            <div
              className="c9"
            >
              <p
                className="typo-element"
              >
                Innhold:
              </p>
              <div>
                <p
                  className="typo-normal"
                >
                  Animi qui molestiae omnis reiciendis quam in. Provident magnam dolor. Omnis voluptatem placeat labore in ut aliquam perspiciatis fugiat. Perspiciatis voluptatum odit.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c7"
          >
            <p
              className="typo-ingress"
            >
              Spørsmål fra bruker - Arbeid
            </p>
            <div
              className="c8"
            >
              <div>
                <p
                  className="typo-element"
                >
                  Kanal: NAV_NO
                </p>
                <p
                  className="typo-element"
                >
                  Type: 
                  SPORSMAL_SKRIFTLIG
                </p>
                <p
                  className="typo-element"
                >
                  Temagruppe: 
                  ARBD
                </p>
                <p
                  className="typo-element"
                >
                   Journalført: 
                  21.07.2019
                   
                </p>
                <p
                  className="typo-element"
                >
                  Journalført med team: 
                </p>
              </div>
              <div>
                <p
                  className="typo-element"
                >
                  Fødselsnummer: 
                  10108000398
                </p>
                <p
                  className="typo-normal"
                >
                  Mottatt/Sendt: 
                  16.07.2019
                </p>
                <p
                  className="typo-normal"
                >
                  Lest: 
                  14.07.2019 00:00
                </p>
              </div>
            </div>
            <div
              className="c9"
            >
              <p
                className="typo-element"
              >
                Innhold:
              </p>
              <div>
                <p
                  className="typo-normal"
                >
                  Quo magni eaque et libero minima ipsum amet et dolorem. Sed iusto ratione sit. Voluptate alias qui impedit ut dolorum doloribus qui. Qui beatae rerum minima suscipit.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c7"
          >
            <p
              className="typo-ingress"
            >
              Delsvar - Arbeid
            </p>
            <div
              className="c8"
            >
              <div>
                <p
                  className="typo-element"
                >
                  Kanal: NAV_NO
                </p>
                <p
                  className="typo-element"
                >
                  Type: 
                  DELVIS_SVAR_SKRIFTLIG
                </p>
                <p
                  className="typo-element"
                >
                  Temagruppe: 
                  ARBD
                </p>
                <p
                  className="typo-element"
                >
                   Journalført: 
                  30.07.2019
                   
                </p>
                <p
                  className="typo-element"
                >
                  Journalført med team: 
                </p>
              </div>
              <div>
                <p
                  className="typo-element"
                >
                  Fødselsnummer: 
                  10108000398
                </p>
                <p
                  className="typo-normal"
                >
                  Mottatt/Sendt: 
                  27.07.2019
                </p>
                <p
                  className="typo-normal"
                >
                  Lest: 
                  23.07.2019 00:00
                </p>
              </div>
            </div>
            <div
              className="c9"
            >
              <p
                className="typo-element"
              >
                Innhold:
              </p>
              <div>
                <p
                  className="typo-normal"
                >
                  Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <p
          aria-live="assertive"
          className="typo-normal"
        >
          Totalt 1 melding
        </p>
      </div>
    </article>
    <h3
      className="sr-only"
      id="Helt tilfeldig ID"
    >
      Meldingsliste - 
      Totalt 1 melding
    </h3>
    <ol
      aria-labelledby="Helt tilfeldig ID"
      className="c11"
      tabIndex={-1}
    >
      <li
        className="c12"
      >
        <input
          checked={true}
          className="sr-only test-meldinger-melding valgt_melding  "
          id="Helt tilfeldig ID"
          name="traadliste-meldinger"
          onChange={[Function]}
          type="radio"
          value="sg838exr"
        />
        <label
          className="c13"
          htmlFor="Helt tilfeldig ID"
        >
          <div
            className="c14"
          >
            <div
              className="c15"
            >
              <span
                className="sr-only"
              >
                (
                4
                )
              </span>
              <span
                className="c16"
              >
                <svg
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#3e3832"
                  >
                    <path
                      d="M16.5 8.75c.476 0 .941.045 1.399.115.065-.379.101-.771.101-1.18C18 3.585 13.962.25 9 .25c-4.963 0-9 3.335-9 7.435 0 1.962.855 3.826 2.36 5.178l-1.794 3.14a.5.5 0 00.62.712l4.839-1.937c.435.124.887.218 1.348.281.022.002.065-.005.075.004.075 0 .319-.002.678-.022.679-3.514 4.233-6.291 8.374-6.291z"
                    />
                    <path
                      d="M16.5 9.75c-4.065 0-7.5 2.977-7.5 6.5 0 3.355 2.878 6.224 6.434 6.572a7.581 7.581 0 003.578-.534l3.813 1.43a.498.498 0 00.623-.691l-1.325-2.65C23.327 19.289 24 17.868 24 16.25c0-3.523-3.435-6.5-7.5-6.5z"
                    />
                  </g>
                </svg>
                <p
                  aria-hidden="true"
                  className="typo-undertekst c17"
                >
                  4
                </p>
              </span>
              <div
                className="c18"
              >
                <div
                  className="c19 order-first"
                >
                  <p
                    className="typo-element order-second"
                  >
                    Delsvar - Arbeid
                  </p>
                  <p
                    className="typo-normal order-first"
                  >
                    27.07.2019 00:00
                  </p>
                </div>
                <div
                  className="c20 order-third"
                >
                  <div
                    className="etikett etikett--fokus"
                  >
                    <span
                      className="typo-normal"
                    >
                      Under arbeid
                    </span>
                  </div>
                  <div
                    className="etikett etikett--info"
                  >
                    <span
                      className="typo-normal"
                    >
                      Delvis besvart
                    </span>
                  </div>
                </div>
                <p
                  className="typo-normal c21 order-second"
                >
                  Dolores omnis ea voluptatem corrupti odit incidunt enim. Unde fugiat sed animi explicabo maiores. Nihil cum accusamus omnis officiis deserunt est. Qui quisquam unde aut magni dignissimos.
                </p>
              </div>
            </div>
          </div>
          <div
            className="c22 hover-animation"
          >
            <i
              className="nav-frontend-chevron chevronboks chevron--hoyre"
            />
          </div>
        </label>
      </li>
    </ol>
  </div>
</nav>
`;
